lua/entities/ent_jack_gmod_ezsolargenerator.lua

```diff
@@ -0,0 +1,247 @@
+                local HitSky = util.TraceLine({start = StartPos, endpos = StartPos + Dir * 9e9, filter = {self}, mask = MASK_SOLID}).HitSky
+                if (HitSky) then HitAmount = HitAmount + 1 end
+                --JMod.Sploom(game.GetWorld(), StartPos + Dir * 1000, 0.5)
+            end
+        end
+        --print(HitAmount)
+        return HitAmount
+    end
+    
+    function ENT:TurnOn()
+        if (self:CheckSky() > 0) then
+            self:EmitSound("buttons/button1.wav", 60, 80)
+            self:SetState(STATE_ON)
+            self.NextUse = CurTime() + 1
+        else
+            self:EmitSound("buttons/button2.wav", 60, 100)
+        end
+    end
+
+    function ENT:TurnOff()
+        self:EmitSound("buttons/button18.wav", 60, 80)
+        self:SetState(STATE_OFF)
+        self:SetProgress(0)
+        self.NextUse = CurTime() + 1
+    end
+    
+    function ENT:Think()
+        local State = self:GetState()
+        if(State == STATE_ON)then
+            --stormfox support
+            if(StormFox)then 
+                if StormFox.GetWeather() ~= "Clear" or StormFox.IsNight() then self:NextThink(CurTime() + 30) return end
+            end
+            if(self.NightMap)then return end
+
+            self:SetVisibility(self:CheckSky())
+            local vis = self:GetVisibility()*0.01
+            local Grade = self:GetGrade()
+            if (vis <= 0 or self:WaterLevel() >= 2) then
+                self:ProducePower()
+                self:TurnOff()
+                return
+            elseif self:GetProgress() < self.MaxPower then 
+                local rate = math.Round((3.34 * Grade * vis), 2)
+                --print(tostring(rate))
+                self:SetProgress(self:GetProgress() + rate)
+                if self:GetProgress() >= 100 then
+                    self:ProducePower()
+                end
+            end
+            --print("Progress: "..self:GetProgress())
+            self:NextThink(CurTime() + 10)
+            return true
+            
+        end
+        
+    end
+elseif(CLIENT)then
+    function ENT:Initialize()
+		self.SolarCellModel = JMod.MakeModel(self,"models/hunter/plates/plate3x5.mdl","models/mat_jack_gmod_solarcells",.5)
+        self.PanelBackModel = JMod.MakeModel(self,"models/hunter/plates/plate3x5.mdl","models/props_pipes/pipeset_metal02",.5)
+        self.ChargerModel = JMod.MakeModel(self,"models/props_lab/powerbox01a.mdl", nil, .5)
+	end
+    local GradeColors={Vector(.3,.3,.3),Vector(.2,.2,.2),Vector(.2,.2,.2),Vector(.2,.2,.2),Vector(.2,.2,.2)}
+	local GradeMats={Material("phoenix_storms/metal"),Material("models/mat_jack_gmod_copper"),Material("models/mat_jack_gmod_silver"),Material("models/mat_jack_gmod_gold"),Material("models/mat_jack_gmod_platinum")}
+    function ENT:Draw()
+		local SelfPos,SelfAng,State=self:GetPos(),self:GetAngles(),self:GetState()
+		local Up,Right,Forward=SelfAng:Up(),SelfAng:Right(),SelfAng:Forward()
+        local Grade = self:GetGrade()
```

Comment:
idk if github is displaying the indents wrong but uh

Reply:
*Shrug* I'll take a look.